plugins/mimecast/komand_mimecast/util/api.py

```diff
@@ -179,17 +179,33 @@ def _handle_zip_file(self, request: requests.request) -> List[Dict]:
             List[Dict]: A list of dictionaries, where each dictionary represents a log entry extracted from the ZIP file
         """
         try:
-            content = self.sanitise_content(request.content)
-            with ZipFile(BytesIO(content)) as my_zip:
-                combined_json_list = [
-                    log for file_name in my_zip.namelist() for log in json.loads(my_zip.read(file_name)).get("data")
-                ]
+            with ZipFile(BytesIO(request.content)) as my_zip:
+                combined_json_list = []
+                for file_name in my_zip.namelist():
+                    self.logger.info(f"About to read file name {file_name}")
+                    try:
+                        contents = my_zip.read(file_name)
+                        for log in json.loads(contents).get("data"):
+                            combined_json_list += [log]
+                    except json.decoder.JSONDecodeError as json_error:
+                        self.logger.error(
```

Comment:
Can you include the filename in here as well? I know it gets logged in the info log line above but it would be useful to have all the info we need in one log line.

Reply:
Updated to include filename 👍